### PR TITLE
Reset value to an empty string when undefined or null

### PR DESCRIPTION
**Pull request by felixlheureux**
Undefined and null values are not properly handled. When an undefined value is passed in Monaco, the textModel will call factory.create on an undefined value which will throw an undefined error. 

Also, if a null value is passed, since the logic returns without updating the value, the previous data will still be displayed.

A solution is to reset the value to an empty string.

Issue: https://github.com/microsoft/monaco-editor/issues/3414

### CODE REVIEW

File: src/vs/editor/common/model/textModel.ts [hunk 1]
```diff
@@ -429,9 +429,8 @@ export class TextModel extends Disposable implements model.ITextModel, IDecorati
 
 	public setValue(value: string | model.ITextSnapshot): void {
 		this._assertNotDisposed();
-		if (value === null) {
-			// There's nothing to do
-			return;
+		if (value === null || value === undefined) {
+			value = '';
```
Comment (alexdima) [inline src/vs/editor/common/model/textModel.ts:433]: The `setValue` function should throw an error (`IllegalArgument`) when invoked with `null` or `undefined`. The caller is violating the contract defined in the argument's type: `string | model.ITextSnapshot`, so the caller of `setValue(undefined)` has a bug.

**Review by alexdima — CHANGES_REQUESTED** [review]
Thanks for pointing this out. The fix should be to throw when an invalid argument is passed in.

File: src/vs/editor/common/model/textModel.ts (hunk 1 above)
Comment (felixlheureux) [reply src/vs/editor/common/model/textModel.ts]: Thank you for the clarification.

**Review by alexdima — APPROVED** [review]

**Review by alexr00 — APPROVED** [review]

**Comment by alexdima** [comment]
Thank you!

**Pull request merged**